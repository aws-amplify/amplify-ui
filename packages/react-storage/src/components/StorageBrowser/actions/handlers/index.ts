export * from './copy';
export * from './createFolder';
export * from './delete';
export * from './download';
export * from './listLocationItems';
export * from './listLocations';
export * from './upload';
export * from './utils';
export * from './types';
