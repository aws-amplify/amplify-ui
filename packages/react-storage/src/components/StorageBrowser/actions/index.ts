export {
  ActionConfigs,
  ActionConfigsProvider,
  ActionConfigsProviderProps,
  ComponentName,
  defaultActionConfigs,
  DefaultActionConfigs,
  DefaultActionKey,
  SelectionType,
  TaskActionConfig,
  useActionConfig,
} from './configs';

export {
  ActionInputConfig,
  copyHandler,
  CopyHandler,
  CopyHandlerData,
  CopyHandlerInput,
  CopyHandlerOutput,
  createFileDataItem,
  createFolderHandler,
  CreateFolderHandler,
  CreateFolderHandlerData,
  CreateFolderHandlerInput,
  CreateFolderHandlerOptions,
  CreateFolderHandlerOutput,
  downloadHandler,
  DownloadHandler,
  DownloadHandlerData,
  DownloadHandlerInput,
  DownloadHandlerOptions,
  DownloadHandlerOutput,
  DeleteHandler,
  DeleteHandlerData,
  DeleteHandlerInput,
  DeleteHandlerOptions,
  DeleteHandlerOutput,
  FileData,
  FileDataItem,
  FileItem,
  FolderData,
  isFileDataItem,
  isFileItem,
  ListHandler,
  ListHandlerInput,
  ListHandlerOptions,
  ListHandlerOutput,
  listLocationItemsHandler,
  ListLocationItemsHandler,
  ListLocationItemsHandlerInput,
  ListLocationItemsHandlerOptions,
  ListLocationItemsHandlerOutput,
  listLocationsHandler,
  ListLocationsHandler,
  ListLocationsHandlerInput,
  ListLocationsHandlerOptions,
  ListLocationsHandlerOutput,
  LocationData,
  LocationItemData,
  LocationType,
  TaskData,
  TaskHandler,
  TaskHandlerInput,
  TaskHandlerOptions,
  TaskHandlerOutput,
  uploadHandler,
  UploadHandler,
  UploadHandlerData,
  UploadHandlerInput,
  UploadHandlerOptions,
  UploadHandlerOutput,
} from './handlers';
