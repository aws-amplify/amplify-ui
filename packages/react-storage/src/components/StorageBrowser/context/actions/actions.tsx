import React from 'react';

import { DataAction } from '@aws-amplify/ui-react-core';

import { Permission } from '../types';
import { useGetLocationConfig } from '../config';

import {
  ActionState,
  createActionStateContext,
} from './createActionStateContext';
import { downloadAction } from './downloadAction';

import { createFolderAction } from './createFolderAction';
import { listLocationItemsAction } from './listLocationItemsAction';
import { ListLocationsAction } from './listLocationsAction';
import { LocationsDataProvider } from './locationsData';

type ActionsWithConfig = {
  [K in keyof DefaultActions]: WithLocationConfig<DefaultActions[K]>;
};

type DefaultActions = typeof DEFAULT_ACTIONS;
type WithLocationConfig<T> = T extends DataAction<infer K, infer U>
  ? DataAction<K, Omit<U, 'config'>>
  : never;

type UseActionState<T> = T extends DataAction<infer K, infer U>
  ? ActionState<K, U>
  : never;

export const ERROR_MESSAGE =
  '`useAction` must be called from within `StorageBrowser.Provider`';

export const DEFAULT_ACTIONS = {
  CREATE_FOLDER: createFolderAction,
  DOWNLOAD: downloadAction,
  LIST_LOCATION_ITEMS: listLocationItemsAction,
};

export const INITIAL_VALUE = {
  CREATE_FOLDER: { result: undefined },
  LIST_LOCATION_ITEMS: { result: [], nextToken: undefined },
  DOWNLOAD: { key: '' },
};

const [ActionStateProvider, useActionState] = createActionStateContext(
  DEFAULT_ACTIONS,
  ERROR_MESSAGE
);

export const useAction = <T extends keyof ActionsWithConfig>({
  type,
}: {
  type: T;
}): UseActionState<ActionsWithConfig[T]> => {
  const [state, handle] = useActionState({ type });
  const getConfig = useGetLocationConfig();
  const handleAction = React.useCallback(
    (input: Parameters<UseActionState<ActionsWithConfig[T]>[1]>) =>
      handle({ ...input, config: getConfig }),
    [getConfig, handle]
  );

  return [state, handleAction] as UseActionState<ActionsWithConfig[T]>;
};

export function ActionProvider<T = Permission>({
  children,
  listLocationsAction,
}: {
  children?: React.ReactNode;
  listLocationsAction: ListLocationsAction<T>;
}): React.JSX.Element {
  return (
    <LocationsDataProvider listLocationsAction={listLocationsAction}>
      <ActionStateProvider initialValue={INITIAL_VALUE}>
        {children}
      </ActionStateProvider>
    </LocationsDataProvider>
  );
}
