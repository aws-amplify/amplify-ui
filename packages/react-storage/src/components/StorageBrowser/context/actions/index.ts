export { createListLocationsAction } from './listLocationsAction';
export { downloadAction } from './downloadAction';
export { ActionProvider, useAction } from './actions';
export { useLocationsData, LocationsDataState } from './locationsData';
export {
  LocationData,
  LocationItem,
  Permission,
  UploadItemData,
} from '../types';
