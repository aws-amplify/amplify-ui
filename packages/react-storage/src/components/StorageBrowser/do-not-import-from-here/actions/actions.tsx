import React from 'react';

import { AsyncDataAction, DataAction } from '@aws-amplify/ui-react-core';

import {
  ActionState,
  InitialValue,
  createActionStateContext,
} from './createActionStateContext';

import { createFolderAction } from './createFolderAction';
import { listLocationItemsAction } from './listLocationItemsAction';
import { ListLocationsAction } from './listLocationsAction';
import { LocationsDataProvider } from './locationsData';
import { useGetActionInput } from '../../providers/configuration';

export type ActionsWithConfig = {
  [K in keyof DefaultActions]: WithLocationConfig<DefaultActions[K]>;
};

export type DefaultActions = typeof DEFAULT_ACTIONS;
export type WithLocationConfig<T> = T extends AsyncDataAction<infer K, infer U>
  ? AsyncDataAction<K, Omit<U, 'config'>>
  : T extends DataAction<infer K, infer U>
  ? DataAction<K, Omit<U, 'config'>>
  : never;

export type UseActionState<T> = T extends
  | AsyncDataAction<infer K, infer U>
  | DataAction<infer K, infer U>
  ? ActionState<K, U>
  : never;

export const ERROR_MESSAGE =
  '`useAction` must be called from within `StorageBrowser.Provider`';

export const DEFAULT_ACTIONS = {
  CREATE_FOLDER: createFolderAction,
  LIST_LOCATION_ITEMS: listLocationItemsAction,
};

export const INITIAL_VALUE: InitialValue<ActionsWithConfig> = {
  CREATE_FOLDER: { result: undefined },
  LIST_LOCATION_ITEMS: { result: [], nextToken: undefined },
};

const [ActionStateProvider, useActionState] = createActionStateContext(
  DEFAULT_ACTIONS,
  ERROR_MESSAGE
);

export const useAction = <T extends keyof ActionsWithConfig>(
  type: T
): UseActionState<ActionsWithConfig[T]> => {
  const [state, handle] = useActionState({ type });

  const getConfig = useGetActionInput();

  const handleAction = React.useCallback(
    (input: Parameters<UseActionState<ActionsWithConfig[T]>[1]>) => {
      const { credentials: credentialsProvider, ...config } = getConfig();
      return handle({ ...input, config: { ...config, credentialsProvider } });
    },
    [getConfig, handle]
  );

  return [state, handleAction] as UseActionState<ActionsWithConfig[T]>;
};

export function ActionProvider({
  children,
  listLocationsAction,
}: {
  children?: React.ReactNode;
  listLocationsAction: ListLocationsAction;
}): React.JSX.Element {
  return (
    <LocationsDataProvider listLocationsAction={listLocationsAction}>
      <ActionStateProvider initialValue={INITIAL_VALUE}>
        {children}
      </ActionStateProvider>
    </LocationsDataProvider>
  );
}
