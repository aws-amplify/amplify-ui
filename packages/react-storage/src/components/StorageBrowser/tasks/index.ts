export { INITIAL_STATUS_COUNTS } from './constants';
export { useProcessTasks } from './useProcessTasks';
export { StatusCounts, Task, Tasks, TaskStatus } from './types';
