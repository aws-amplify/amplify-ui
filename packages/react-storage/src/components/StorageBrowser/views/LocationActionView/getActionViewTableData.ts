import { humanFileSize } from '@aws-amplify/ui';

import { DataTableProps } from '../../composables/DataTable';
import { Task } from '../../tasks';

import {
  DEFAULT_ACTION_VIEW_HEADERS,
  PROGRESS_HEADER,
  STATUS_DISPLAY_VALUES,
} from './constants';

import { isFileItem, isFileDataItem } from '../../actions';
import { getActionIcon } from './getActionIcon';
import { getFileTypeDisplayValue } from './getFileTypeDisplayValue';
import { getPercentValue } from '../utils';

export const getActionViewTableData = ({
  tasks,
  locationKey,
  isProcessing,
  shouldDisplayProgress = false,
  onTaskCancel,
}: {
  tasks: Task[];
  locationKey?: string;
  isProcessing: boolean;
  shouldDisplayProgress?: boolean;
  onTaskCancel: (task: Task) => void;
}): DataTableProps => {
  const headers = [...DEFAULT_ACTION_VIEW_HEADERS];
  if (shouldDisplayProgress) {
    headers.splice(-1, 0, PROGRESS_HEADER);
  }

  const rows: DataTableProps['rows'] = tasks.map((task) => {
    const { data, progress, status } = task;
    const { id } = data;
    const displayKey = isFileDataItem(data)
      ? data.fileKey
      : data.key.split('/').pop() ?? '';

    return {
      key: id,
      content: headers.map(({ key: columnKey }) => {
        const key = `${columnKey}-${id}`;
        switch (columnKey) {
          case 'name': {
            return {
              key,
              type: 'text',
              content: {
                icon: getActionIcon(status),
                text: displayKey,
              },
            };
          }
          case 'folder': {
            if (locationKey) {
              return { key, type: 'text', content: { text: locationKey } };
            }

            if (isFileItem(data)) {
              const { webkitRelativePath } = data.file;
              return {
                key,
                type: 'text',
                content: {
                  text: webkitRelativePath
                    ? webkitRelativePath.slice(
                        0,
                        webkitRelativePath.lastIndexOf('/') + 1
                      )
                    : '-',
                },
              };
            }

            return { key, type: 'text', content: { text: '/' } };
          }
          case 'type': {
            return {
              key,
              type: 'text',
              content: { text: getFileTypeDisplayValue(displayKey) },
            };
          }
          case 'size': {
            const value = isFileItem(data)
              ? data.file.size
              : isFileDataItem(data)
              ? data.size
              : undefined;
            return {
              key,
              type: 'number',
              content: {
                value,
                displayValue: value ? humanFileSize(value, true) : '-',
              },
            };
          }
          case 'status': {
            return {
              key,
              type: 'text',
              content: { text: STATUS_DISPLAY_VALUES[status] },
            };
          }
          case 'progress': {
            return {
              key,
              type: 'number',
              content: {
                value: progress,
                displayValue: `${getPercentValue(progress ?? 0)}%`,
              },
            };
          }
          case 'cancel': {
            const isDisabled =
              (isProcessing && !task.cancel) ||
              (status !== 'PENDING' && status !== 'QUEUED');
            const ariaLabel = `${
              isProcessing ? 'Cancel' : 'Remove'
            } item: ${displayKey}`;

            return {
              key,
              type: 'button',
              content: {
                isDisabled,
                onClick: () => {
                  onTaskCancel(task);
                },
                ariaLabel,
                icon: 'cancel',
              },
            };
          }

          default:
            return { key, type: 'text', content: {} };
        }
      }),
    };
  });

  return { headers, rows };
};
