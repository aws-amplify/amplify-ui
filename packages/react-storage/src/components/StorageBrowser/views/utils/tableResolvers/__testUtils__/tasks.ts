import { MULTIPART_UPLOAD_THRESHOLD_BYTES } from '../../../../actions/handlers/constants';
import { TaskStatus } from '../../../../tasks';

import { CopyActionTask, DeleteActionTask, UploadActionTask } from '../types';

type MockCopyOrDeleteTaskStatus = Exclude<TaskStatus, 'OVERWRITE_PREVENTED'>;

type MockCopyTasks = {
  [K in MockCopyOrDeleteTaskStatus]: Omit<CopyActionTask, 'status'> & {
    status: K;
  };
};

type MockDeleteTasks = {
  [K in MockCopyOrDeleteTaskStatus]: Omit<DeleteActionTask, 'status'> & {
    status: K;
  };
};

type MockUploadTasks = {
  [K in TaskStatus]: Omit<UploadActionTask, 'status'> & { status: K };
};

const MOCK_PREFIX = 'test-folder/';
const MOCK_COPY_OR_DELETE_TASK_BASE = {
  fileKey: 'file1.txt',
  lastModified: new Date(),
  size: 1000,
  type: 'FILE' as const,
};

const MOCK_COPY_DESTINATION_PREFIX = 'some-destination-folder/';
const MOCK_COPY_DATA: Omit<CopyActionTask['data'], 'id'> = {
  ...MOCK_COPY_OR_DELETE_TASK_BASE,
  key: `${MOCK_COPY_DESTINATION_PREFIX}${MOCK_PREFIX}file1.txt`,
  sourceKey: `${MOCK_PREFIX}file1.txt`,
};

export const MOCK_COPY_TASKS: MockCopyTasks = {
  CANCELED: {
    data: {
      ...MOCK_COPY_DATA,
      id: 'CANCELED-ID',
    },
    cancel: undefined,
    message: undefined,
    progress: undefined,
    status: 'CANCELED',
  },
  FAILED: {
    data: {
      ...MOCK_COPY_DATA,
      id: 'FAILED-ID',
    },
    cancel: undefined,
    status: 'FAILED',
    message: undefined,
    progress: undefined,
  },
  COMPLETE: {
    data: {
      ...MOCK_COPY_DATA,
      id: 'COMPLETE-ID',
    },
    cancel: undefined,
    status: 'COMPLETE',
    message: undefined,
    progress: undefined,
  },
  QUEUED: {
    data: {
      ...MOCK_COPY_DATA,
      id: 'QUEUED-ID',
    },
    cancel: jest.fn(),
    status: 'QUEUED',
    message: undefined,
    progress: undefined,
  },
  PENDING: {
    data: {
      ...MOCK_COPY_DATA,
      id: 'PENDING-ID',
    },
    cancel: undefined,
    status: 'PENDING',
    message: undefined,
    progress: undefined,
  },
};

const MOCK_DELETE_DATA: Omit<DeleteActionTask['data'], 'id'> = {
  ...MOCK_COPY_OR_DELETE_TASK_BASE,
  key: `${MOCK_PREFIX}file1.txt`,
};

export const MOCK_DELETE_TASKS: MockDeleteTasks = {
  CANCELED: {
    data: {
      ...MOCK_DELETE_DATA,
      id: 'CANCELED-ID',
    },
    cancel: undefined,
    message: undefined,
    progress: undefined,
    status: 'CANCELED',
  },
  FAILED: {
    data: {
      ...MOCK_DELETE_DATA,
      id: 'FAILED-ID',
    },
    status: 'FAILED',
    message: undefined,
    progress: undefined,
  },
  COMPLETE: {
    data: {
      ...MOCK_DELETE_DATA,
      id: 'COMPLETE-ID',
    },
    status: 'COMPLETE',
    message: undefined,
    progress: undefined,
  },
  QUEUED: {
    data: {
      ...MOCK_DELETE_DATA,
      id: 'QUEUED-ID',
    },
    cancel: jest.fn(),
    status: 'QUEUED',
    message: undefined,
    progress: undefined,
  },
  PENDING: {
    data: {
      ...MOCK_DELETE_DATA,
      id: 'PENDING-ID',
    },
    status: 'PENDING',
    message: undefined,
    progress: undefined,
  },
};

// single part upload
const SMALL_FILE = {
  name: 'SMALL_FILE.jpg',
  size: 100,
  webkitRelativePath: 'local-folder/',
} as File;
const SMALL_FILE_KEY = `target-prefix/${SMALL_FILE.name}`;

export const MOCK_UPLOAD_TASKS_SINGLE_PART: MockUploadTasks = {
  CANCELED: {
    data: { id: 'CANCELED-ID', key: SMALL_FILE_KEY, file: SMALL_FILE },
    cancel: undefined,
    message: undefined,
    progress: 0.8,
    status: 'CANCELED',
  },
  COMPLETE: {
    data: { id: 'COMPLETE-ID', key: SMALL_FILE_KEY, file: SMALL_FILE },
    cancel: undefined,
    message: undefined,
    progress: 1,
    status: 'COMPLETE',
  },
  FAILED: {
    data: { id: 'FAILED-ID', key: SMALL_FILE_KEY, file: SMALL_FILE },
    cancel: undefined,
    message: 'some error message',
    progress: 0.8,
    status: 'FAILED',
  },
  OVERWRITE_PREVENTED: {
    data: {
      id: 'OVERWRITE_PREVENTED-ID',
      key: SMALL_FILE_KEY,
      file: SMALL_FILE,
    },
    cancel: undefined,
    message: 'some error message',
    progress: 1,
    status: 'OVERWRITE_PREVENTED',
  },
  PENDING: {
    data: { id: 'PENDING-ID', key: SMALL_FILE_KEY, file: SMALL_FILE },
    cancel: jest.fn(),
    message: undefined,
    progress: 0.8,
    status: 'PENDING',
  },
  QUEUED: {
    data: { id: 'QUEUED-ID', key: SMALL_FILE_KEY, file: SMALL_FILE },
    cancel: jest.fn(),
    message: undefined,
    progress: undefined,
    status: 'QUEUED',
  },
};

// multipart upload
const LARGE_FILE = {
  name: 'LARGE_FILE.mpeg',
  size: 1.1 * MULTIPART_UPLOAD_THRESHOLD_BYTES,
  webkitRelativePath: 'local-folder/',
} as File;
const LARGE_FILE_KEY = `target-prefix/${LARGE_FILE.name}`;

export const MOCK_UPLOAD_TASKS_MULTIPART = Object.entries(
  MOCK_UPLOAD_TASKS_SINGLE_PART
).reduce(
  (items, [status, item]) => ({
    ...items,
    [status]: {
      ...item,
      data: { ...item.data, file: LARGE_FILE, key: LARGE_FILE_KEY },
    },
  }),
  {} as MockUploadTasks
);
