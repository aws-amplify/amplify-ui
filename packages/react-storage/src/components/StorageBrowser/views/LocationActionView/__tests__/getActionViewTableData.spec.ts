import { FileDataItem } from '../../../actions/handlers';
import { Tasks } from '../../../tasks';

import { getActionViewTableData } from '../getActionViewTableData';

describe('getActionViewTableData', () => {
  it('should return correct table data for all task statuses', () => {
    const tasks: Tasks<FileDataItem> = [
      {
        data: {
          id: '1',
          key: 'some-prefix/file1.txt',
          fileKey: 'file1.txt',
          lastModified: new Date(),
          size: 1000,
          type: 'FILE',
        },
        status: 'QUEUED',
        cancel: jest.fn(),
        message: '',
        progress: undefined,
      },
      {
        data: {
          id: '2',
          key: 'some-prefix/file2.jpg',
          fileKey: 'file2.jpg',
          lastModified: new Date(),
          size: 1000,
          type: 'FILE',
        },
        status: 'PENDING',
        cancel: jest.fn(),
        message: '',
        progress: undefined,
      },
      {
        data: {
          id: '3',
          key: 'some-prefix/file3.pdf',
          fileKey: 'file3.pdf',
          lastModified: new Date(),
          size: 1000,
          type: 'FILE',
        },
        status: 'COMPLETE',
        cancel: jest.fn(),
        message: '',
        progress: undefined,
      },
      {
        data: {
          id: '4',
          key: 'some-prefix/file4.doc',
          fileKey: 'file4.doc',
          lastModified: new Date(),
          size: 1000,
          type: 'FILE',
        },
        status: 'FAILED',
        cancel: jest.fn(),
        message: '',
        progress: undefined,
      },
      {
        data: {
          id: '5',
          key: 'some-prefix/file5',
          fileKey: 'file5',
          lastModified: new Date(),
          size: 1000,
          type: 'FILE',
        },
        status: 'CANCELED',
        cancel: jest.fn(),
        message: '',
        progress: undefined,
      },
    ];
    const result = getActionViewTableData({
      tasks,
      isProcessing: true,
      onTaskRemove: jest.fn(),
    });

    expect(result.rows).toMatchSnapshot('tabledata');
  });

  it('should handle tasks with prefix keys', () => {
    const tasks: Tasks<FileDataItem> = [
      {
        data: {
          id: '1',
          key: 'folder/subfolder/file1.txt',
          fileKey: 'file1.txt',
          lastModified: new Date(),
          size: 1000,
          type: 'FILE',
        },
        status: 'QUEUED',

        cancel: jest.fn(),
        progress: undefined,
        message: '',
      },
      {
        data: {
          id: '2',
          key: '/root/file2.jpg',
          fileKey: 'file2.jpg',
          lastModified: new Date(),
          size: 1000,
          type: 'FILE',
        },
        status: 'COMPLETE',

        cancel: jest.fn(),
        message: '',
        progress: undefined,
      },
    ];

    const result = getActionViewTableData({
      tasks,
      isProcessing: true,
      onTaskRemove: jest.fn(),
    });

    expect(result.rows).toMatchSnapshot();
  });

  it('should have remove handler on queued files', () => {
    const tasks: Tasks<FileDataItem> = [
      {
        data: {
          id: '1',
          fileKey: 'file1.txt',
          key: 'folder/subfolder/file1.txt',
          lastModified: new Date(),
          size: 1000,
          type: 'FILE',
        },
        status: 'QUEUED',

        cancel: jest.fn(),
        progress: undefined,
        message: '',
      },
    ];

    const result = getActionViewTableData({
      tasks,
      locationKey: 'folder/subfolder/',
      isProcessing: false,
      onTaskRemove: jest.fn(),
    });

    // last cell
    const actionCell =
      result.rows[0].content[result.rows[0].content.length - 1];

    expect(actionCell.content).toHaveProperty('onClick');
    expect(actionCell.content).toHaveProperty(
      'ariaLabel',
      'Remove item: file1.txt'
    );
    expect(result.rows).toMatchSnapshot();
  });
});
