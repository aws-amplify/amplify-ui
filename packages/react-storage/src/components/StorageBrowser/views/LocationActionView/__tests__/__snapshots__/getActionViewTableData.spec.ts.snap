// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getActionViewTableData should handle tasks with prefix keys 1`] = `
[
  {
    "content": [
      {
        "content": {
          "icon": "action-progress",
          "text": "file1.txt",
        },
        "key": "name-1",
        "type": "text",
      },
      {
        "content": {
          "text": "/",
        },
        "key": "folder-1",
        "type": "text",
      },
      {
        "content": {
          "text": "txt",
        },
        "key": "type-1",
        "type": "text",
      },
      {
        "content": {
          "displayValue": "1.0 kB",
          "value": 1000,
        },
        "key": "size-1",
        "type": "number",
      },
      {
        "content": {
          "text": "In progress",
        },
        "key": "status-1",
        "type": "text",
      },
      {
        "content": {
          "ariaLabel": "Cancel item: file1.txt",
          "icon": "cancel",
          "isDisabled": false,
          "onClick": [Function],
        },
        "key": "cancel-1",
        "type": "button",
      },
    ],
    "key": "1",
  },
  {
    "content": [
      {
        "content": {
          "icon": "action-success",
          "text": "file2.jpg",
        },
        "key": "name-2",
        "type": "text",
      },
      {
        "content": {
          "text": "/",
        },
        "key": "folder-2",
        "type": "text",
      },
      {
        "content": {
          "text": "jpg",
        },
        "key": "type-2",
        "type": "text",
      },
      {
        "content": {
          "displayValue": "1.0 kB",
          "value": 1000,
        },
        "key": "size-2",
        "type": "number",
      },
      {
        "content": {
          "text": "Completed",
        },
        "key": "status-2",
        "type": "text",
      },
      {
        "content": {
          "ariaLabel": "Cancel item: file2.jpg",
          "icon": "cancel",
          "isDisabled": true,
          "onClick": [Function],
        },
        "key": "cancel-2",
        "type": "button",
      },
    ],
    "key": "2",
  },
]
`;

exports[`getActionViewTableData should have remove handler on queued files 1`] = `
[
  {
    "content": [
      {
        "content": {
          "icon": "action-queued",
          "text": "file1.txt",
        },
        "key": "name-1",
        "type": "text",
      },
      {
        "content": {
          "text": "folder/subfolder/",
        },
        "key": "folder-1",
        "type": "text",
      },
      {
        "content": {
          "text": "txt",
        },
        "key": "type-1",
        "type": "text",
      },
      {
        "content": {
          "displayValue": "1.0 kB",
          "value": 1000,
        },
        "key": "size-1",
        "type": "number",
      },
      {
        "content": {
          "text": "Not started",
        },
        "key": "status-1",
        "type": "text",
      },
      {
        "content": {
          "ariaLabel": "Remove item: file1.txt",
          "icon": "cancel",
          "isDisabled": false,
          "onClick": [Function],
        },
        "key": "cancel-1",
        "type": "button",
      },
    ],
    "key": "1",
  },
]
`;

exports[`getActionViewTableData should return correct table data for all task statuses: tabledata 1`] = `
[
  {
    "content": [
      {
        "content": {
          "icon": "action-queued",
          "text": "file1.txt",
        },
        "key": "name-1",
        "type": "text",
      },
      {
        "content": {
          "text": "/",
        },
        "key": "folder-1",
        "type": "text",
      },
      {
        "content": {
          "text": "txt",
        },
        "key": "type-1",
        "type": "text",
      },
      {
        "content": {
          "displayValue": "1.0 kB",
          "value": 1000,
        },
        "key": "size-1",
        "type": "number",
      },
      {
        "content": {
          "text": "Not started",
        },
        "key": "status-1",
        "type": "text",
      },
      {
        "content": {
          "ariaLabel": "Cancel item: file1.txt",
          "icon": "cancel",
          "isDisabled": false,
          "onClick": [Function],
        },
        "key": "cancel-1",
        "type": "button",
      },
    ],
    "key": "1",
  },
  {
    "content": [
      {
        "content": {
          "icon": "action-progress",
          "text": "file2.jpg",
        },
        "key": "name-2",
        "type": "text",
      },
      {
        "content": {
          "text": "/",
        },
        "key": "folder-2",
        "type": "text",
      },
      {
        "content": {
          "text": "jpg",
        },
        "key": "type-2",
        "type": "text",
      },
      {
        "content": {
          "displayValue": "1.0 kB",
          "value": 1000,
        },
        "key": "size-2",
        "type": "number",
      },
      {
        "content": {
          "text": "In progress",
        },
        "key": "status-2",
        "type": "text",
      },
      {
        "content": {
          "ariaLabel": "Cancel item: file2.jpg",
          "icon": "cancel",
          "isDisabled": false,
          "onClick": [Function],
        },
        "key": "cancel-2",
        "type": "button",
      },
    ],
    "key": "2",
  },
  {
    "content": [
      {
        "content": {
          "icon": "action-success",
          "text": "file3.pdf",
        },
        "key": "name-3",
        "type": "text",
      },
      {
        "content": {
          "text": "/",
        },
        "key": "folder-3",
        "type": "text",
      },
      {
        "content": {
          "text": "pdf",
        },
        "key": "type-3",
        "type": "text",
      },
      {
        "content": {
          "displayValue": "1.0 kB",
          "value": 1000,
        },
        "key": "size-3",
        "type": "number",
      },
      {
        "content": {
          "text": "Completed",
        },
        "key": "status-3",
        "type": "text",
      },
      {
        "content": {
          "ariaLabel": "Cancel item: file3.pdf",
          "icon": "cancel",
          "isDisabled": true,
          "onClick": [Function],
        },
        "key": "cancel-3",
        "type": "button",
      },
    ],
    "key": "3",
  },
  {
    "content": [
      {
        "content": {
          "icon": "action-error",
          "text": "file4.doc",
        },
        "key": "name-4",
        "type": "text",
      },
      {
        "content": {
          "text": "/",
        },
        "key": "folder-4",
        "type": "text",
      },
      {
        "content": {
          "text": "doc",
        },
        "key": "type-4",
        "type": "text",
      },
      {
        "content": {
          "displayValue": "1.0 kB",
          "value": 1000,
        },
        "key": "size-4",
        "type": "number",
      },
      {
        "content": {
          "text": "Failed",
        },
        "key": "status-4",
        "type": "text",
      },
      {
        "content": {
          "ariaLabel": "Cancel item: file4.doc",
          "icon": "cancel",
          "isDisabled": true,
          "onClick": [Function],
        },
        "key": "cancel-4",
        "type": "button",
      },
    ],
    "key": "4",
  },
  {
    "content": [
      {
        "content": {
          "icon": "action-canceled",
          "text": "file5",
        },
        "key": "name-5",
        "type": "text",
      },
      {
        "content": {
          "text": "/",
        },
        "key": "folder-5",
        "type": "text",
      },
      {
        "content": {
          "text": "",
        },
        "key": "type-5",
        "type": "text",
      },
      {
        "content": {
          "displayValue": "1.0 kB",
          "value": 1000,
        },
        "key": "size-5",
        "type": "number",
      },
      {
        "content": {
          "text": "Canceled",
        },
        "key": "status-5",
        "type": "text",
      },
      {
        "content": {
          "ariaLabel": "Cancel item: file5",
          "icon": "cancel",
          "isDisabled": true,
          "onClick": [Function],
        },
        "key": "cancel-5",
        "type": "button",
      },
    ],
    "key": "5",
  },
]
`;
